hotpdf/encodings/decoder.py

```diff
@@ -0,0 +1,33 @@
+import re
+from typing import Optional
+
+from .types import EncodingTypes
+
+
+class Decoder:
+    """If there are no embedded fonts, pdfminer.six has issues with
+    mapping cid values to their specific unicode characters.
+    In that case, we manually override the (cid:int) values
+    """
+
+    __cid_mapping: dict[int, str] = {}
+    initialised: bool = False
+
+    def __init__(self, charset: Optional[EncodingTypes] = None) -> None:
+        if not charset:
+            return
+        if charset.value in EncodingTypes._value2member_map_:
+            self.initialised = True
+        if charset == EncodingTypes.LATIN:
+            from .mappings.latin import CID_TO_STR
+
+            self.__cid_mapping = CID_TO_STR
```

Comment:
Possibly after this construction we might check if nothing was initialized throw an error Unitialized decoder, just for ensuring we didn't skip the logical case

Reply:
That makes sense, I will add it 👍